### DISCUSSION TITLE
Using ProjectData function with Seurat v5

### DISCUSSION THREAD

**Discussion by rjmusich** [satijalab/seurat#7586] [Q&A]
I was following along with the "Sketch integration using a 1 million cell dataset from Parse Biosciences" vignette for Seurat v5 and was able to run everything successfully, however I wondered about the call to ProjectData in the vignette and was wondering if there was a typo. In the vignette, it uses the following parameters: sketched.assay = 'sketch', assay = 'RNA', sketched.reduction = 'integrated.rpca.full', & full.reduction = 'integrated.rpca.full'. If the 'sketched.reduction' is looking for the dimensional reduction that was performed on the 'sketch' assay, should it not be using the 'integrated.rpca' reduction that was generated using the 'sketch' assay? For reference, the 'integrated.rpca.full' reduction is generated in the line before that runs ProjectIntegration.

I ran the ProjectData function once using 'integrated.rpca' and another using 'integrated.rpca.full' and I cannot visually tell a difference between the results so it could just be that due to the projection of the 'sketch' dataset onto the full dataset, the 'integration.rpca' & 'integrated.rpca.full' end up having generally the same information to pull from. Other than this, I was able to follow the rest of the vignette and understand how everything works so this was the only thing that jumped out to me as being odd.

**Comment by Dario-Rocha** [comment]
I have doubts regardin a similar issue when integrating sketched data and the projecting integration and projecting data. Apparently I am following the same vignette you are (https://satijalab.org/seurat/articles/parsebio_sketch_integration), but in this other vignette (https://satijalab.org/seurat/articles/seurat5_sketch_analysis.html) the use of ProjectData function seems to be a bit different:
```
obj <- [ProjectData](https://satijalab.org/seurat/reference/ProjectData.html)(
  object = obj,
  assay = "RNA",
  full.reduction = "pca.full",
  sketched.assay = "sketch",
  sketched.reduction = "pca",
  umap.model = "umap",
  dims = 1:50,
  refdata = [list](https://rdrr.io/r/base/list.html)(cluster_full = "seurat_clusters")
)
```
In this second vignette, we can see that they calculated the UMAP on the sketched reduction and then projected the full data onto this UMAP. In the first vignette, instead, they project the full data onto the sketched RPCA and then calculate the UMAP on that projected RPCA. What alarmed me is that in my case, the UMAP calculated on the integrated.rpca.full looks absolutely nothing like the umap calculated on the sketched data integrated.rpca

**Reply by rjmusich** [reply]
Yeah, I actually combed through these a bit to figure out what the differences were and they are actually not as dissimilar as they may seem. The code you have posted here is from a vignette that did not require integration of the sketched assay, so in this case there is no 'integrated.rpca.full' and really the only reduction to bring over from the sketched assay is the 'pca' and the 'umap' reductions that were calculated on the sketched assay.

In the other vignette (the one referenced in my original post), you do the extra integration step on the sketched assay, which creates the 'integrated.rpca' and the ProjectIntegration then takes the 'sketched.reduction' of 'integrated.rpca' and maps it back to the full dataset and names that reduction as 'integrated.rpca.full', which I don't believe is actually re-running a 'rpca' analysis but just adding plotting coordinates to the 'non-sketch' cells based on their proximity in expression to 'sketch' cells. In this vignette, they also re-calculate the UMAP, which is probably why yours looks a bit different than the original 'sketch' UMAP. When I ran my data through, the position of some of the clusters changed somewhat, but the general cluster definition and proximity to neighboring clusters was still the same.

The reason the one you posted probably looks the same is that by skipping the 'ProjectIntegration' step, you can just add the 'umap.model = "umap" ' and it just automatically keeps the same UMAP coordinates of the 'sketch' assay and essentially wraps the remaining cells around that UMAP. In short, without integration, it uses the 'sketch' UMAP as a template to build of of, but, with integration, you kind of through out the template and just re-cluster.

One thing I did notice through trial-and-error with the whole integration of the 'sketch' assay was that trying to project a 'sketch' dataset onto a full dataset that contained a sample of very low-quality completely blew up the resulting 'full' UMAP. So it is probably advised to filter out REALLY bad samples (like >50% dead cells) prior to doing the projection...
